sources/types.ts

```diff
@@ -122,3 +136,7 @@ export interface Locator {
      */
   reference: string;
 }
+export interface URLLocator {
+  name: string;
+  reference: URL;
```

Comment:
I'd avoid working with `URL` instances, as it complexifies significantly the code for little value. We can as well keep them a string and have a `isUrl` util that simply checks whether they start with `https?`.

Reply:
Done in https://github.com/nodejs/corepack/pull/359/commits/146ba91e401dc366e6d7f5b76ee5f5a1e2180157

> as it complexifies significantly the code for little value

Maybe I'm missing something, but I don't find the new implementation less complex, rather the opposite 🤔